lib/Sema/CSDiag.cpp

```diff
@@ -4264,8 +4264,14 @@ bool TypeChecker::diagnoseSelfAssignment(const Expr *E) {
   if (!AE)
     return false;
 
+  // Allow the use of parentheses to express self-assignment when it's a deliberate choice
+  auto src = AE->getSrc();
+  auto loadSrc = dyn_cast<LoadExpr>(src);
+  if (loadSrc != nullptr && isa<ParenExpr>(loadSrc->getSubExpr()))
```

Comment:
We generally just do `if (p && p->...)` rather than `if (p != nullptr && p->...)` but it doesn't look like it's enshrined in the coding standards anywhere.